fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/lao/CreateLao.java

```diff
@@ -29,10 +31,18 @@ public class CreateLao extends Data {
    * @param creation time of creation
    * @param organizer id of the LAO's organizer
    * @param witnesses list of witnesses of the LAO
-   * @throws IllegalArgumentException if the id is not valid
+   * @throws IllegalArgumentException if arguments are invalid
    */
   public CreateLao(
-      String id, String name, long creation, PublicKey organizer, List<PublicKey> witnesses) {
+      @NonNull String id,
+      @NonNull String name,
+      long creation,
+      @NonNull PublicKey organizer,
+      @NonNull List<PublicKey> witnesses) {
+    // organizer and witnesses are checked to be base64 at deserialization
+    checkBase64(id, "id");
+    checkStringNotEmpty(name, "name");
+    checkValidTime(creation);
```

Comment:
Probably the only suggestion would be a more structured approach. A [template pattern](https://refactoring.guru/design-patterns/template-method) could be a solution. Having an abstract class with a factory method (could be `checkValidity`) which calls all the small methods that perform a punctual check and extend it with a class that overrides each of them in place too, so it's more extensible for the future

Reply:
Sounds good, I will look into it!